### ISSUE TITLE
Why don't we use `dataclasses.dataclass` or `pydantic` to define pythonic datamodels, instead of the custom `Serializable` ABC?

### ISSUE THREAD

**Issue by kwsp** [waltsims/uff.py#28]
Why don't we use `dataclasses.dataclass` or `pydantic` to define pythonic datamodels, instead of the custom `Serializable` ABC?

Two issues: 

1. `Serializable`'s definition implies that it's an abstract base class, yet it's not exactly "abstract" and provides the actual implementation of helper methods `serialize` and `deserialize`, which makes it a ["mix-in"](https://stackoverflow.com/questions/533631/what-is-a-mixin-and-why-is-it-useful) class rather than an ABC.
2. From reading the code, it seems like the functionalities provided by `Serializable` can be accomplished by the standard library `dataclass` or (if desiring something more powerful), the [`pydantic`](https://pydantic-docs.helpmanual.io/) classes.

Just some thoughts and rambling since the `Serializable` mechanism doesn't seem to be very robust as I'm getting bugs from `k-wave-python`

**Comment by kwsp** [comment]
`pydantic` should be pretty good at handling nested serialization/deserialization https://pydantic-docs.helpmanual.io/usage/exporting_models/

IIRC the standard library `dataclass` doesn't handle nested deserialization

**Comment by waltsims** [comment]
Yes thank you for this. Thinking back the nested serialization was the reasons for not using data classes in the first place. A rewrite would make sense.

**Comment by kwsp** [comment]
I would help when I get some free time. I actually don't know anything about the "Ultrasound File Format" but it looks very interesting. Definitely nice to have a standard for sharing data.
